### ISSUE TITLE
Setting SHOW_TRIGGER_FORM_IF_NO_PARAMS=True redirecting to trigger form for every DAG instead of giving an option to select as in 2.6.3

### ISSUE THREAD

**Issue by vatsrahul1001** [apache/airflow#33392]
### Apache Airflow version
airflow main branch 

### What happened

While Testing [PR](https://github.com/apache/airflow/pull/33351), i noticed after setting `AIRFLOW__WEBSERVER__SHOW_TRIGGER_FORM_IF_NO_PARAMS =True`, even the DAG which does not have Params or dag_run.conf is getting redirecting automatically to trigger w/ config form 

### What you think should happen instead

As per me this should give the option to trigger with conf or not instead to redirecting automatically as we should to have in 2.6.3In

### How to reproduce

Run breeze with latest main set `AIRFLOW__WEBSERVER__SHOW_TRIGGER_FORM_IF_NO_PARAMS=True`  and try to trigger any DAG its automatically redirecting to trigger config form

### Operating System

Mac os 

### Deployment

Astronomer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Thanks for the report. This has already been addressed and the main reason why we cancelled RC1 voting and we are going to have RC2.

**Comment by vatsrahul1001** [comment]
@potiukthis bug is against the [PR](https://github.com/apache/airflow/pull/33351) was which was fix against the [bug](https://github.com/apache/airflow/issues/33344) i raised earlier,setting SHOW_TRIGGER_FORM_IF_NO_PARAMS=True should go back to original behavior as mentioned [here](https://github.com/apache/airflow/pull/33351#issuecomment-1676121764), however, its automatically redirecting to trigger page for every DAG which is not right

**Comment by ephraimbuddy** [comment]
> Thanks for the report. This has already been addressed and the main reason why we cancelled RC1 voting and we are going to have RC2.

@potiuk , this seems different. When the config is enabled, every dag when triggered opens up the trigger page but previously, you have the option of choosing to trigger with config or not.

**Comment by potiuk** [comment]
Ah. OK Sorry then.

**Comment by jens-scheffler-bosch** [comment]
The statement I tried to make in the "hof fix PR"  was meant to get to get "functional" on the same level not exactly re-producing the exact visual. Main point was that it was not able to trigger a DAG with config when no params are defined - and ths the form automatically be skipped.
It was not expressed that it is "critical" to replicate the same popup, thus being able to skip w/o config also if params are defined.

I originally proposed multiple options from which the voted option was implemented. The hotfix was intended to re-establish the option for users to force a config. Reverting all the visual is more than a small change, I'd propose to re-discuss the final "target" behavior, seems to be multiple opinions around what a trigger button should do and what not. One or the other way, two click are needed to trigger.
